test/compress/issue-203.js

```diff
@@ -26,9 +26,7 @@ compress_new_function_with_destruct: {
     }
     expect: {
         Function("a", "[b]", "return a");
-        Function("a", "{bb}", "return a");
-        Function("[[a]]", "[{bb}]", 'return a');
+        Function("a", "{bb:b}", "return a");
+        Function("[[a]]", "[{bb:b}]", 'return a');
```

Comment:
> I see. Interesting. Although I did not see any mention of `side_effects` needing to be disabled there. Maybe it's for the best it's not active by default.

There is no special treatment of `side_effects` within that optimisation rule - the interaction is unintentional, and I'm comtemplating at the moment whether it is also sub-optimal.